### PR TITLE
Added namespace support.

### PR DESCRIPTION
**Pull request by bkuhlmann**
## Overview

Ran into a deployment issue when attempting to add Rake namespaces (for organizational purposes) to our apps. This will prevent deployment failures because the hooks were not properly defined within the namespace.
## Details
- See commits for details.

### CODE REVIEW

File: lib/kapost_deploy/task.rb [hunk 1]
```diff
@@ -87,21 +87,28 @@ def define
...
+      @name = name
+      @scope = scope
     end
 
     attr_accessor :plugins
+    attr_reader :scope
+
+    def hook_name(type)
+      return "#{name}:#{type}_#{name}" if scope.empty?
+      "#{scope}:#{name}:#{type}_#{name}"
```
Comment (paul) [inline lib/kapost_deploy/task.rb:101]:
``` ruby
task = [type, name].join("_")
[scope, name, task].compact.join(":")
```

Comment (bkuhlmann) [inline lib/kapost_deploy/task.rb:101]: Thanks, I had considered that but the problem is that when the namespace is missing, it answers an empty string instead so `#compact` won't catch it.

I have a [refinement](https://github.com/bkuhlmann/refinements#array) for this if we wanted to add a gem dependency that would make this more elegant if you're up for it?

Comment (paul) [inline lib/kapost_deploy/task.rb:101]: Nah, lets not add another dependency. This is fine.

We could also have done `.delete_if { |el| el.blank? }`

**Comment by paul** [comment]
Also, don't bump the gem version in PRs. Easier to just do it on master at release time.

File: lib/kapost_deploy/task.rb (hunk 1 above)
Comment (bkuhlmann) [inline lib/kapost_deploy/task.rb:101]: True...not as performant, though.

**Comment by bkuhlmann** [comment]
ok, thanks. Merging and deploying new version of gem.

**Pull request merged**